src/Extractor/LinkExtractorInterface.php

```diff
@@ -0,0 +1,14 @@
+<?php
+/**
+ * @license   http://opensource.org/licenses/BSD-3-Clause BSD-3-Clause
+ * @copyright Copyright (c) 2014 Zend Technologies USA Inc. (http://www.zend.com)
+ */
+
+namespace ZF\Hal\Extractor;
+
+use Zend\Stdlib\Extractor\ExtractionInterface;
```

Comment:
A specific interface could be interesting in order to add type hinting in extract method declaration and avoid manual parameter validation.